notebooks/official/tabular_workflows/tabnet_on_vertex_pipelines.ipynb

```diff
@@ -0,0 +1,1315 @@
+{
```

Comment:
<ul><li>Do we have definitions for these parameters?</li><li>What's the difference between this pipeline and the one before it? </li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/726/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

Reply:
<p>The section before creates a TabNet CustomJob pipeline and this one creates a TabNet HyperparameterTuningJob pipeline. The interface and API we call are a bit different so we have examples for both. The hyperparameters in <code>parameters</code> are defined in the  <code>Training inputs</code> section under  <code>Model hyperparameters</code>. I put the definitions up there because the hyperparameters are common to both the CustomJob pipeline and the HyperparameterTuningJob pipeline, but are set in different ways.</p>

Reply:
<p>I think it's normal that the interface is different for training and tuning. For training the hyperparameter is a fixed value since there is nothing to tune. For training the hyparameter is a range (described by double_value_spec, integer_value_spec, categorical_value_spec, and discrete_value_spec).</p><p> </p>

Reply:
<p>Oh, sorry, I thought double_value_spec is a parameter name. Why do we say "The available parameter spec types are double_value_spec, integer_value_spec, categorical_value_spec, and discrete_value_spec." in the introduction? If I understand correctly a double_value_spec is a parameter that accepts a double as the parameter value. Isn't it better to say that directly in the description of the associated parameter? </p>

Reply:
<p>I still don't understand what you do with a CustomJob configuration and a HyperparameterTuningJob configuration. A CustomJob configuration allows you to customize training machine type and specify the <code>learning_rate</code>,&nbsp;<code>max_steps</code>,&nbsp;<code>max_train_secs</code>hyperparameters? And then the HyperpameterTuningJob also allows you to customize training machine type and specify hyperparameters, but the set of hyperparameters is bigger and contained in a separate "parameters" object? </p><p> </p><p>Is it possible to run "automatic" TabNet, accepting all the default options? </p><p> </p><p>Can we modify the <code>learning_rate</code>,&nbsp;<code>max_steps</code>,&nbsp;<code>max_train_secs</code>hyperparameters in the HyperpameterTuningJob? </p>

Reply:
That's right, but I see the confusion. `learning_rate`, `max_steps`, and `max_train_secs` are required inputs for a CustomJob. They're also required for a HyperparameterTuningJob, but contained within the `parameters` list and we have a helper function to return that. I'll remove the "customize" wording. It's confusing.